reframe/core/logging.py

```diff
@@ -678,39 +678,35 @@ def emit(self, record):
             raise LoggingError('logging failed') from e
 
 
+_create_handler_registry = {
+    'file': _create_file_handler,
+    'filelog': _create_filelog_handler,
+    'syslog': _create_syslog_handler,
+    'stream': _create_stream_handler,
+    'graylog': _create_graylog_handler,
+    'httpjson': _create_httpjson_handler,
+}
+
+
+def register_plugin_handler(create_plugin_handler):
+    _create_handler_registry["plugin"] = create_plugin_handler
```

Comment:
I would make this a decorator taking the name of the handler to register that will populate the above registry upon definition of each handler. This registry could start empty here and every log handler creation function (even the builtin ones) could be defined as

```python
@register_log_handler('handler_name')
def _create_foo_handler(site_config, handlers_group):
    ...
```

The nice thing with the decorator is that we can also do some validation of the function signature if we want to.

```suggestion
def register_log_handler(name):
    def _deco(fn):
        _create_handler_registry[name] = fn
        return fn
     
     return deco
```